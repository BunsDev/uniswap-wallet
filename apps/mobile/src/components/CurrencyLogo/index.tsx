import React from 'react'
import { TokenLogo } from 'src/components/CurrencyLogo/TokenLogo'
import { iconSizes } from 'ui/src/theme/iconSizes'
import { CurrencyInfo } from 'wallet/src/features/dataApi/types'

interface CurrencyLogoProps {
  currencyInfo: Maybe<CurrencyInfo>
  size?: number
  hideNetworkLogo?: boolean
}

export const STATUS_RATIO = 2 / 5

export function CurrencyLogo({
  currencyInfo,
  size = iconSizes.icon40,
  hideNetworkLogo,
}: CurrencyLogoProps): JSX.Element | null {
  if (!currencyInfo) return null

  const { currency, logoUrl } = currencyInfo
  const { chainId, symbol } = currency

  return (
    <TokenLogo
      chainId={chainId}
      hideNetworkLogo={hideNetworkLogo}
      size={size}
      symbol={symbol}
      url={logoUrl}
    />
  )
}
