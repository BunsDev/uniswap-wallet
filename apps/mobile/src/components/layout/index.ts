export * from './Box'
export * from './DeprecatedMobileOnlyFlex'
export * from './Inset'
