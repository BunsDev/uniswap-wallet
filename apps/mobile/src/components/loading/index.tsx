import React, { memo, useMemo } from 'react'
import { NftCardLoader } from 'src/components/loading/NftCardLoader'
import { TokenLoader } from 'src/components/loading/TokenLoader'
import { TransactionLoader } from 'src/components/loading/TransactionLoader'
import { WalletLoader } from 'src/components/loading/WalletLoader'
import { WaveLoader } from 'src/components/loading/WaveLoader'
import { Flex, getToken } from 'ui/src'
import { BoxLoader, BoxLoaderProps, Shimmer } from 'ui/src/loading'

function Graph(): JSX.Element {
  return (
    <Shimmer>
      <WaveLoader />
    </Shimmer>
  )
}

function Wallets({ repeat = 1 }: { repeat?: number }): JSX.Element {
  return (
    <Shimmer>
      <Flex gap="$spacing12">
        {new Array(repeat).fill(null).map((_, i, { length }) => (
          <React.Fragment key={i}>
            <WalletLoader opacity={(length - i) / length} />
          </React.Fragment>
        ))}
      </Flex>
    </Shimmer>
  )
}

function Token({ repeat = 1, contrast }: { repeat?: number; contrast?: boolean }): JSX.Element {
  return (
    <Shimmer contrast={contrast}>
      <Flex grow gap="$spacing4">
        {new Array(repeat).fill(null).map((_, i, { length }) => (
          <React.Fragment key={i}>
            <TokenLoader opacity={(length - i) / length} />
          </React.Fragment>
        ))}
      </Flex>
    </Shimmer>
  )
}

export const Transaction = memo(function _Transaction({
  repeat = 1,
}: {
  repeat?: number
}): JSX.Element {
  return (
    <Shimmer>
      <Flex>
        {new Array(repeat).fill(null).map((_, i, { length }) => (
          <React.Fragment key={i}>
            <TransactionLoader opacity={(length - i) / length} />
          </React.Fragment>
        ))}
      </Flex>
    </Shimmer>
  )
})

function NFT({ repeat = 1 }: { repeat?: number }): JSX.Element {
  const loader = useMemo(
    () =>
      repeat === 1 ? (
        <NftCardLoader opacity={1} />
      ) : (
        <Flex>
          {new Array(repeat / 2).fill(null).map((_, i) => {
            const firstColOpacity = (repeat - ((repeat / 2) * i + 1) + 1) / repeat
            const secondColOpacity = (repeat - ((repeat / 2) * i + 2) + 1) / repeat
            return (
              <React.Fragment key={i}>
                <Flex row>
                  <NftCardLoader opacity={firstColOpacity} width="50%" />
                  <NftCardLoader opacity={secondColOpacity} width="50%" />
                </Flex>
              </React.Fragment>
            )
          })}
        </Flex>
      ),
    [repeat]
  )

  return <Shimmer>{loader}</Shimmer>
}

function Box(props: BoxLoaderProps): JSX.Element {
  return (
    <Shimmer>
      <BoxLoader {...props} />
    </Shimmer>
  )
}

function Image(): JSX.Element {
  return (
    <Shimmer>
      <BoxLoader aspectRatio={1} borderRadius={getToken('$none', 'radius')} />
    </Shimmer>
  )
}

function Favorite({ height, contrast }: { height?: number; contrast?: boolean }): JSX.Element {
  return (
    <Shimmer contrast={contrast}>
      <BoxLoader backgroundColor="$surface3" borderRadius="$rounded16" height={height ?? 50} />
    </Shimmer>
  )
}

export const Loader = {
  Box,
  NFT,
  Token,
  Transaction,
  Wallets,
  Graph,
  Image,
  Favorite,
}
