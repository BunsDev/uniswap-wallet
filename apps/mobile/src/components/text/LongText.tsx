import React, { ComponentProps, useCallback, useReducer, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { LayoutChangeEvent, NativeSyntheticEvent, TextLayoutEventData } from 'react-native'
import Markdown, { MarkdownProps } from 'react-native-markdown-display'
import {
  runOnJS,
  useAnimatedReaction,
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
} from 'react-native-reanimated'
import { openUri } from 'src/utils/linking'
import { AnimatedFlex, Flex, SpaceTokens, Text, useSporeColors } from 'ui/src'
import { fonts } from 'ui/src/theme'

type LongTextProps = {
  initialDisplayedLines?: number
  text: string
  gap?: SpaceTokens
  color?: string
  linkColor?: string
  codeBackgroundColor?: string
  readMoreOrLessColor?: string
  renderAsMarkdown?: boolean
  variant?: keyof typeof fonts
} & Omit<
  ComponentProps<typeof Text>,
  'children' | 'numberOfLines' | 'onTextLayout' | 'color' | 'variant'
>

export function LongText(props: LongTextProps): JSX.Element {
  const colors = useSporeColors()
  const { t } = useTranslation()
  const {
    initialDisplayedLines = 3,
    text,
    gap = '$spacing8',
    color = colors.neutral1.get(),
    linkColor = colors.neutral2.get(),
    readMoreOrLessColor = colors.neutral2.get(),
    renderAsMarkdown = false,
    codeBackgroundColor = colors.surface3.get(),
    variant = 'body2',
    ...rest
  } = props

  const [expanded, toggleExpanded] = useReducer(
    (isExpanded) => !isExpanded,
    renderAsMarkdown ? true : false
  )
  const [textLengthExceedsLimit, setTextLengthExceedsLimit] = useState(false)

  const initialContentHeight = useSharedValue<number | undefined>(undefined)
  const textLineHeight = useSharedValue(0)
  const maxVisibleHeight = useDerivedValue(
    () => textLineHeight.value * initialDisplayedLines,
    [initialDisplayedLines]
  )
  const animatedStyle = useAnimatedStyle(
    () => ({
      height: !textLengthExceedsLimit || expanded ? 'auto' : maxVisibleHeight.value,
      overflow: 'hidden',
    }),
    [expanded, textLengthExceedsLimit]
  )

  useAnimatedReaction(
    () => ({
      initialHeight: initialContentHeight.value,
      lineHeight: textLineHeight.value,
    }),
    ({ initialHeight, lineHeight }) => {
      if (initialHeight === undefined) return false
      const numberOfLines = Math.floor(initialHeight / lineHeight)
      runOnJS(setTextLengthExceedsLimit)(numberOfLines >= initialDisplayedLines)
    }
  )

  const onMarkdownLayout = useCallback(
    (event: LayoutChangeEvent) => {
      if (initialContentHeight.value !== undefined) return
      initialContentHeight.value = event.nativeEvent.layout.height
      toggleExpanded()
    },
    [initialContentHeight]
  )

  const onTextLayout = useCallback(
    (e: NativeSyntheticEvent<TextLayoutEventData>) => {
      setTextLengthExceedsLimit(e.nativeEvent.lines.length >= initialDisplayedLines)
    },
    [initialDisplayedLines]
  )

  const codeStyle = { backgroundColor: codeBackgroundColor, borderColor: 'transparent' }

  const markdownStyle: MarkdownProps['style'] = {
    body: {
      color,
      fontFamily: fonts[variant].family,
      overflow: 'hidden',
    },
    code_inline: codeStyle,
    fence: codeStyle,
    code_block: codeStyle,
    link: { color: linkColor },
    paragraph: {
      marginBottom: 0,
      marginTop: 0,
      fontSize: fonts[variant].fontSize,
      lineHeight: fonts[variant].lineHeight,
    },
  }

  return (
    <Flex gap={gap}>
      {renderAsMarkdown ? (
        <AnimatedFlex style={animatedStyle} onLayout={onMarkdownLayout}>
          {/* Render fake one-line markdown to properly measure the height of a single text line */}
          <Flex
            opacity={0}
            pointerEvents="none"
            position="absolute"
            onLayout={({
              nativeEvent: {
                layout: { height },
              },
            }): void => {
              textLineHeight.value = height
            }}>
            <Markdown
              style={{
                ...markdownStyle,
                body: {},
                paragraph: { ...markdownStyle.paragraph, lineHeight: fonts[variant].lineHeight },
              }}
              {...{ children: '.' }}
            />
          </Flex>
          <Markdown
            style={markdownStyle}
            onLinkPress={(url): false => {
              // add our own custom link handler since it has security checks that only open http/https links
              openUri(url).catch(() => undefined)
              return false
            }}
            // HACK: children prop no in TS definition
            {...{ children: text }}
          />
        </AnimatedFlex>
      ) : (
        <Text
          numberOfLines={expanded ? undefined : initialDisplayedLines}
          style={{ color }}
          variant={variant}
          onTextLayout={onTextLayout}
          {...rest}>
          {text}
        </Text>
      )}

      {/* Text is removed vs hidden using opacity to ensure spacing after the element is consistent in all cases.
      This will cause mild thrash as data loads into a page but will ensure consistent spacing */}
      {textLengthExceedsLimit ? (
        <Text
          my="$none"
          py="$none"
          style={{ color: readMoreOrLessColor }}
          testID="read-more-button"
          variant="buttonLabel3"
          onPress={toggleExpanded}>
          {expanded ? t('Read less') : t('Read more')}
        </Text>
      ) : null}
    </Flex>
  )
}
