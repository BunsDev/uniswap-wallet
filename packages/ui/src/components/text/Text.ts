import { styled, Text as TamaguiText } from 'tamagui'
import { fonts } from 'ui/src/theme/fonts'

export const Text = styled(TamaguiText, {
  name: 'Text',

  // TODO(EXT-61): keep investigating how to get text to wrap
  wordWrap: 'break-word',
  flex: 1,
  flexGrow: 0, // Would expect the default to be this, but default seems to be 1
  flexWrap: 'wrap',

  variants: {
    // TODO: leverage font tokens instead
    // https://tamagui.dev/docs/core/configuration#font-tokens
    // https://tamagui.dev/docs/core/font-language
    variant: {
      headlineLarge: {
        fontFamily: '$heading',
        fontSize: fonts.headlineLarge.fontSize,
        lineHeight: fonts.headlineLarge.lineHeight,
        fontWeight: '600',
      },
      headlineMedium: {
        fontFamily: '$heading',
        fontSize: fonts.headlineMedium.fontSize,
        lineHeight: fonts.headlineMedium.lineHeight,
        fontWeight: '500',
      },
      headlineSmall: {
        fontFamily: '$heading',
        fontSize: fonts.headlineSmall.fontSize,
        lineHeight: fonts.headlineSmall.lineHeight,
        fontWeight: '500',
      },
      subheadLarge: {
        fontFamily: '$heading',
        fontSize: fonts.subheadLarge.fontSize,
        lineHeight: fonts.subheadLarge.lineHeight,
        fontWeight: '500',
      },
      subheadSmall: {
        fontFamily: '$heading',
        fontSize: fonts.subheadSmall.fontSize,
        lineHeight: fonts.subheadSmall.lineHeight,
        fontWeight: '500',
      },
      bodyLarge: {
        fontFamily: '$body',
        fontSize: fonts.bodyLarge.fontSize,
        lineHeight: fonts.bodyLarge.lineHeight,
        fontWeight: '500',
      },
      bodySmall: {
        fontFamily: '$body',
        fontSize: fonts.bodySmall.fontSize,
        lineHeight: fonts.bodySmall.lineHeight,
        fontWeight: '400',
      },
      bodyMicro: {
        fontFamily: '$body',
        fontSize: fonts.bodyMicro.fontSize,
        lineHeight: fonts.bodyMicro.lineHeight,
        fontWeight: '400',
      },
      buttonLabelLarge: {
        fontFamily: '$body',
        fontSize: fonts.buttonLabelLarge.fontSize,
        lineHeight: fonts.buttonLabelLarge.lineHeight,
        fontWeight: '600',
      },
      buttonLabelMedium: {
        fontFamily: '$body',
        fontSize: fonts.buttonLabelMedium.fontSize,
        lineHeight: fonts.buttonLabelMedium.lineHeight,
        fontWeight: '600',
      },
      buttonLabelSmall: {
        fontFamily: '$body',
        fontSize: fonts.buttonLabelSmall.fontSize,
        lineHeight: fonts.buttonLabelSmall.lineHeight,
        fontWeight: '600',
      },
      buttonLabelMicro: {
        fontFamily: '$body',
        fontSize: fonts.buttonLabelMicro.fontSize,
        lineHeight: fonts.buttonLabelMicro.lineHeight,
        fontWeight: '600',
      },
      monospace: {
        fontFamily: '$body',
        fontSize: fonts.bodySmall.fontSize,
        lineHeight: fonts.bodySmall.lineHeight,
      },
    },
  } as const,
})
