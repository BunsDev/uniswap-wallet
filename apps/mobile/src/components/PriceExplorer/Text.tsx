import React from 'react'
import { SharedValue, useAnimatedStyle } from 'react-native-reanimated'
import { useLineChartDatetime } from 'react-native-wagmi-charts'
import { useAppTheme } from 'src/app/hooks'
import { AnimatedCaretChange } from 'src/components/icons/Caret'
import { Flex } from 'src/components/layout'
import { Text } from 'src/components/Text'
import { AnimatedText } from 'src/components/text/AnimatedText'
import { AnimatedDecimalNumber } from './AnimatedDecimalNumber'
import { useLineChartPrice, useLineChartRelativeChange } from './usePrice'

export function PriceText({ loading }: { loading: boolean }): JSX.Element {
  const price = useLineChartPrice()

  if (loading) {
    return <Text loading loadingPlaceholderText="$10,000" variant="headlineLarge" />
  }

  return <AnimatedDecimalNumber number={price} testID="price-text" variant="headlineLarge" />
}

export function RelativeChangeText({
  loading,
  spotRelativeChange,
}: {
  loading: boolean
  spotRelativeChange?: SharedValue<number>
}): JSX.Element {
  const theme = useAppTheme()

  const relativeChange = useLineChartRelativeChange({ spotRelativeChange })

  const styles = useAnimatedStyle(() => ({
    color:
      relativeChange.value.value > 0 ? theme.colors.accentSuccess : theme.colors.accentCritical,
  }))
  const caretStyle = useAnimatedStyle(() => ({
    color:
      relativeChange.value.value > 0 ? theme.colors.accentSuccess : theme.colors.accentCritical,
    transform: [{ rotate: relativeChange.value.value > 0 ? '180deg' : '0deg' }],
  }))

  if (loading) {
    return <Text loading loadingPlaceholderText="00.00%" variant="bodyLarge" />
  }

  return (
    <Flex row alignItems="flex-end" gap="spacing2">
      <AnimatedCaretChange
        height={theme.iconSizes.icon16}
        strokeWidth={2}
        style={caretStyle}
        width={theme.iconSizes.icon16}
      />
      <AnimatedText
        style={styles}
        testID="relative-change-text"
        text={relativeChange.formatted}
        variant="bodyLarge"
      />
    </Flex>
  )
}

export function DatetimeText({ loading }: { loading: boolean }): JSX.Element | null {
  // `datetime` when scrubbing the chart
  const datetime = useLineChartDatetime()

  if (loading) return null

  return <AnimatedText color="textSecondary" text={datetime.formatted} variant="bodyLarge" />
}
