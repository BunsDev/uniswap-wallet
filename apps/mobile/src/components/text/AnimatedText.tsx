import {
  color,
  ColorProps,
  createRestyleComponent,
  createVariant,
  typography,
  TypographyProps,
  VariantProps,
} from '@shopify/restyle'
import React from 'react'
import {
  StyleSheet,
  TextInput,
  TextInputProps,
  TextProps as RNTextProps,
  useWindowDimensions,
} from 'react-native'
import Animated, { useAnimatedProps } from 'react-native-reanimated'
import { DEFAULT_FONT_SCALE } from 'src/components/Text'
import { Flex } from 'ui/src'
import { TextLoaderWrapper } from 'ui/src/components/text/Text'
import { fonts, TextVariantTokens } from 'ui/src/theme'
import { Theme } from 'ui/src/theme/restyle'

// base animated text component using a TextInput
// forked from https://github.com/wcandillon/react-native-redash/blob/master/src/ReText.tsx
// and modified to support the loading state
Animated.addWhitelistedNativeProps({ text: true })

interface TextProps extends Omit<TextInputProps, 'value' | 'style'> {
  text?: Animated.SharedValue<string>
  style?: Animated.AnimateProps<RNTextProps>['style']
  loading?: boolean | 'no-shimmer'
  loadingPlaceholderText?: string
}

const AnimatedTextInput = Animated.createAnimatedComponent(TextInput)

export const BaseAnimatedText = ({
  style,
  text,
  loading,
  loadingPlaceholderText = '$00.00',
  ...rest
}: TextProps): JSX.Element => {
  const animatedProps = useAnimatedProps(() => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    return {
      text: text?.value,
      // Here we use any because the text prop is not available in the type
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
    } as any
  })

  if (loading) {
    return (
      <TextLoaderWrapper loadingShimmer={loading !== 'no-shimmer'}>
        <Flex row>
          {/* Use empty input for loading shimmer height calculation (it is different
          than the text component height) */}
          <AnimatedTextInput
            editable={false}
            style={[style, styles.loadingInput]}
            underlineColorAndroid="transparent"
            {...rest}
          />
          {/* Use the text component to properly calculate the width of the loading shimmer.
          An input component with a width dependent on the length of the content was sometimes 
          rendered with a very small width regardless of the text passed as a value */}
          <Animated.Text style={[style, styles.loadingPlaceholder]}>
            {loadingPlaceholderText}
          </Animated.Text>
        </Flex>
      </TextLoaderWrapper>
    )
  }

  return (
    <AnimatedTextInput
      editable={false}
      style={[style || undefined]}
      underlineColorAndroid="transparent"
      value={text?.value}
      {...rest}
      {...{ animatedProps }}
    />
  )
}
// end of forked from https://github.com/wcandillon/react-native-redash/blob/master/src/ReText.tsx

// exposes restyle to base animated text
const StyledBaseAnimatedText = createRestyleComponent<
  VariantProps<Theme, 'textVariants'> &
    TypographyProps<Theme> &
    ColorProps<Theme> &
    React.ComponentProps<typeof BaseAnimatedText>,
  Theme
>([createVariant({ themeKey: 'textVariants' }), typography, color], BaseAnimatedText)

// wrapped around restyled animated text with convenience props
export const AnimatedText = (
  props: React.ComponentProps<typeof StyledBaseAnimatedText>
): JSX.Element => {
  const { fontScale } = useWindowDimensions()
  const enableFontScaling = fontScale > DEFAULT_FONT_SCALE

  const variant = (props.variant ?? 'body2') as TextVariantTokens
  const multiplier = fonts[variant].maxFontSizeMultiplier

  return (
    <StyledBaseAnimatedText
      {...props}
      allowFontScaling={enableFontScaling}
      maxFontSizeMultiplier={multiplier}
      style={[styles.input, props.style]}
    />
  )
}

const styles = StyleSheet.create({
  input: {
    padding: 0, // inputs have default padding on Android
  },
  loadingInput: {
    marginHorizontal: 0,
    opacity: 0,
    paddingHorizontal: 0,
    width: 0,
  },
  loadingPlaceholder: {
    opacity: 0,
  },
})
