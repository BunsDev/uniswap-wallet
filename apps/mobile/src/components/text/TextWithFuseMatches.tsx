import Fuse from 'fuse.js'
import React from 'react'

import { Flex } from 'src/components/layout'
import { Text, TextProps } from 'src/components/Text'

interface TextWithFuseMatchesProps {
  text: string
  matches?: readonly Fuse.FuseResultMatch[]
  variant?: TextProps['variant']
  numberOfLines?: TextProps['numberOfLines']
}

export function TextWithFuseMatches({
  matches,
  text,
  variant = 'bodyLarge',
  numberOfLines = 1,
}: TextWithFuseMatchesProps & TextProps): JSX.Element {
  if (!matches || matches.length === 0) {
    return (
      <Text color="neutral1" numberOfLines={numberOfLines} variant={variant}>
        {text}
      </Text>
    )
  }

  const charIsMatch = new Set()
  for (const match of matches) {
    for (const index of match.indices) {
      for (let i = index[0]; i < index[1] + 1; i++) {
        charIsMatch.add(i)
      }
    }
  }

  // PERF: batch pieces?
  const pieces = []
  for (let i = 0; i < text.length; i++) {
    if (charIsMatch.has(i)) {
      pieces.push([text[i], true])
    } else {
      pieces.push([text[i], false])
    }
  }

  const elements = (
    <>
      {pieces.map((p, i) => {
        if (p[1])
          return (
            <Text key={`${i}${p[0]}`} color="neutral1" variant={variant}>
              {p[0]}
            </Text>
          )
        else
          return (
            <Text key={`${i}${p[0]}`} color="neutral3" variant={variant}>
              {p[0]}
            </Text>
          )
      })}
    </>
  )

  return <Flex row>{elements}</Flex>
}
