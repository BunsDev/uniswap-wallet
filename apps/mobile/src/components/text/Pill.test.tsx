import React from 'react'
import { Text } from 'src/components/Text'
import { Pill } from 'src/components/text/Pill'
import { render } from 'src/test/test-utils'

it('renders a Pill without image', () => {
  const tree = render(
    <Pill backgroundColor="surface2" foregroundColor="accent1" label="My Pill Label" />
  )
  expect(tree).toMatchSnapshot()
})

it('renders a Pill with border', () => {
  const tree = render(
    <Pill borderColor="statusSuccess" icon={<Text>Icon</Text>} label="My Second Pill Label" />
  )
  expect(tree).toMatchSnapshot()
})
