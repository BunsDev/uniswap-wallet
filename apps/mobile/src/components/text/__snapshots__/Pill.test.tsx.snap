// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Pill with border 1`] = `
<View
  sentry-label="Flex"
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#40B66B",
        "borderRadius": 999999,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 4,
        "paddingVertical": 8,
      },
      {},
    ]
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    style={
      [
        {
          "display": "flex",
        },
      ]
    }
  >
    Icon
  </Text>
  <View
    style={
      [
        {
          "display": "flex",
          "flexGrow": 0,
          "flexShrink": 0,
          "width": 8,
        },
      ]
    }
  />
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    style={
      [
        {
          "color": "#0E1524",
          "display": "flex",
          "fontFamily": "Inter-Regular",
          "fontSize": 15,
          "lineHeight": 20,
        },
        {
          "color": undefined,
        },
      ]
    }
  >
    My Second Pill Label
  </Text>
</View>
`;

exports[`renders a Pill without image 1`] = `
<View
  sentry-label="Flex"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "transparent",
        "borderRadius": 999999,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 4,
        "paddingVertical": 8,
      },
      {},
    ]
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    style={
      [
        {
          "color": "#0E1524",
          "display": "flex",
          "fontFamily": "Inter-Regular",
          "fontSize": 15,
          "lineHeight": 20,
        },
        {
          "color": "accentActive",
        },
      ]
    }
  >
    My Pill Label
  </Text>
</View>
`;
